src/Reflex/Dom/Widget/Input.hs

```diff
@@ -212,6 +214,110 @@ dropdown k0 options (DropdownConfig setK attrs) = do
   dValue <- combineDyn readKey options =<< holdDyn (Just k0) (leftmost [eChange, fmap Just setK])
   return $ Dropdown dValue (attachDynWith readKey options eChange)
 
+
+data ButtonGroup t a
+   = ButtonGroup { _buttonGroup_value    :: Dynamic t (Maybe a)
+                 , _buttonGroup_select   :: Event  t (Maybe a)
+                 , _buttonGroup_element  :: El t
+                 , _buttonGroup_children :: Dynamic t (Map.Map Int (El t))
```

Comment:
This seems a bit heavy.  Can we allow the children to return a value of their choice rather than always returning the El?  That way we wouldn't have to always wrap things in the El (so it's slightly more performant) and we could also return other stuff (so it's more general).

Also, you shouldn't need to qualify the type `Map`.  We should have these import lines above, if we don't already:
```haskell
import Data.Map (Map)
import qualified Data.Map as Map
```